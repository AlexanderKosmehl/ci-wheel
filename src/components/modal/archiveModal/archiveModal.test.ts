import {
  describe, expect, it, vi,
} from 'vitest';
import generateArchiveModal from './archiveModal';

describe('generateArchiveModal', () => {
  const onClose = vi.fn(() => {});
  const updateCallback = vi.fn((_updatedEntries: string[]) => {});

  const modal = generateArchiveModal({
    onClose,
    currentEntries: ['Test 1', 'Test 2'],
    updateCurrentEntries: updateCallback,
  });

  it('generates component correctly', () => {
    expect(modal).toMatchSnapshot();
  });
});
