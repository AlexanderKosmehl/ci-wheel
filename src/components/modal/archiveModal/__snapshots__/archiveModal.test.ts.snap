// Vitest Snapshot v1

exports[`generateArchiveModal > generates component correctly 1`] = `
<div
  class="container"
>
  <div
    class="content"
  >
    <div
      class="header"
    >
      <h2
        class="headerText"
      >
        Archiv
      </h2>
      <button
        class="button closeButton"
        data-test="modalCloseButton"
      >
        <img
          class="icon"
          src="/src/assets/close-icon.svg"
        />
      </button>
    </div>
    <div
      class="archiveModalContainer"
    >
      <div
        class="column"
      >
        <h3
          class="header"
        >
          Aktuell
        </h3>
        <ul
          class="listContainer"
        >
          <li
            class="entry is-draggable"
          >
            Test 1
          </li>
          <li
            class="entry is-draggable"
          >
            Test 2
          </li>
        </ul>
      </div>
      <div
        class="column"
      >
        <h3
          class="header"
        >
          Alt
        </h3>
        <ul
          class="listContainer"
        />
      </div>
      <div
        class="removalContainer"
      >
        <img
          class="removalImage"
          src="/src/assets/delete-icon.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
